includes/polling/functions.inc.php

```diff
@@ -626,17 +630,43 @@ function json_app_get($device, $extend, $min_version = 1)
 {
     $output = snmp_get($device, 'nsExtendOutputFull.' . string_to_oid($extend), '-Oqv', 'NET-SNMP-EXTEND-MIB');
 
+    // save for returning if not JSON
+    $orig_output = $output;
+
     // make sure we actually get something back
     if (empty($output)) {
         throw new JsonAppPollingFailedException('Empty return from snmp_get.', -2);
     }
 
+    // checks for base64 decoding and converts it to non-base64 so it can gunzip
+    if (preg_match('/^[A-Za-z0-9\/\+\n]+\=*\n*$/', $output)) {
+        $output = base64_decode($output);
+        $i = 10;
+        $flg = ord(substr($output, 3, 1));
+        if ($flg > 0) {
+            if ($flg & 4) {
+                [$xlen] = unpack('v', substr($output, $i, 2));
+                $i = $i + 2 + $xlen;
+            }
+            if ($flg & 8) {
+                $i = strpos($output, "\0", $i) + 1;
+            }
+            if ($flg & 16) {
+                $i = strpos($output, "\0", $i) + 1;
+            }
+            if ($flg & 2) {
+                $i = $i + 2;
+            }
+        }
+        $output = gzinflate(substr($output, $i, -8));
```

Comment:
not sure what all of this does, but why not do something like;

```php
if (
  (($output= base64_decode($output, true)) !== false) 
  &&
  (($output= gzdecode($output)) !== false)
) {
 // unpacked gzip+base64 data
}


```

Reply:
It yanks the header off so gzinflate works on the string in question. http://www.faqs.org/rfcs/rfc1952.html

Just tried gzdecode and that worked with you needing to yank the header, so switched to that. Thanks!